flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/BackPressureStatsTracker.java

```diff
@@ -177,7 +177,7 @@ public boolean triggerStackTraceSample(ExecutionJobVertex vertex) {
 						LOG.debug("Triggering stack trace sample for tasks: " + Arrays.toString(vertex.getTaskVertices()));
 					}
 
-					Future<StackTraceSample> sample = coordinator.triggerStackTraceSample(
+					CompletableFuture<StackTraceSample> sample = coordinator.triggerStackTraceSample(
```

Comment:
The Java `Future` does not allow you to apply transformations to it. That was only introduced with Java 8's `CompletableFuture`.